feat: set locale domains via runtimeConfig (#2443)

cjpearson opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)

Please carefully read the contribution docs before creating a pull request
 👉 https://nuxtjs.com/docs/community/contribution
-->

### 🔗 Linked issue

#2443 

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

This change allows setting locale domains via the runtime config. Currently when enabling `differentDomains` the app must be recompiled for each environment because the domains are set at compile time. With this change a single build can be used on different environments with different .env files.

e.g. 

production .env

```
NUXT_PUBLIC_I18N_LOCALES_EN_DOMAIN=en.example.test
NUXT_PUBLIC_I18N_LOCALES_DE_DOMAIN=de.example.test
```

staging .env
```
NUXT_PUBLIC_I18N_LOCALES_EN_DOMAIN=en.staging.example.test
NUXT_PUBLIC_I18N_LOCALES_DE_DOMAIN=de.staging.example.test
```

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

cjpearson committed:
feat: set locale domains via runtimeConfig (#2443)

cjpearson commented:
Is this test maybe flaky? I can't reproduce it locally and it also seems to be causing the failures in #2442 and https://github.com/nuxt-modules/i18n/commit/ee5e026c7e9ae02837911e94e424ca29dfa3d7da

BobbieGoede commented:
@cjpearson 
Yes the tests are a bit flaky in general (tracked here https://github.com/nuxt-modules/i18n/issues/1998), so will be rerunning it. Fortunately it seems to only throw false failures and not the other way around 😅

BobbieGoede commented:
Thanks for contributing! 

I think this feature looks good, I prefer it over the approach currently described in the docs here: https://v8.i18n.nuxtjs.org/guide/different-domains#runtime-environment-variables

I think if you change/replace the example in the linked documentation to use this feature it would be perfect! What do you think @kazupon?

cjpearson committed:
docs: add example of runtime domain configuration to the guide

cjpearson commented:
Thanks! I made a pass at updating the guide with an example

kazupon commented:
Thank you for your contribution!
I think this feature looks good too.
we would like to merge this feature, when you will ready for example and docs :)

cjpearson commented:
Hi @kazupon I updated the docs for runtime config and routing and added an example to the different domains guide. Is there anything else I should change in the docs?

BobbieGoede commented:
I have raised the test job timeout, if you rebase your branch on `next` the tests should succeed (at least it won't fail because of the timeout)

BobbieGoede commented:
Not sure why the tests keep failing repeatedly now, I know that it's not because of your code as they pass locally. Looks like the flaky test debt has caught up with us 😭, will take some time to look into this.

kazupon commented:
Yeah. 
The Other PRs such as #2441 are also flaky testing… we should fix that issue. 😿
